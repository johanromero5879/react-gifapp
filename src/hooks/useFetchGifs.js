import { useEffect, useState } from "react"
import { getGifs } from "../helpers/getGifs"

export const useFetchGifs = ( category ) => {
    const [state, setState] = useState({
        data: [],
        loading: true
    })

    useEffect(() => {
        const fetchGifs = async () => {
            const gifs = await getGifs(category)
            setState({
                data: gifs,
                loading: false
            })
        }
        fetchGifs()
    }, [ category ])

    return state
}