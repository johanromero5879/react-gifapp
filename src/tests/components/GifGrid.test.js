import { shallow } from 'enzyme'

import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from '../../hooks/useFetchGifs';

jest.mock('../../hooks/useFetchGifs') //falsificar useFetchGifs

describe('Testing with <GifGrid />', () => {
    const category = 'hug anime'
        
    // beforeEach( () => {
    //     jest.clearAllMocks()
    //     wrapper = shallow( <GifGrid category={ category } /> )
    // } )
        

    test('should render correctly', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        })

        const wrapper = shallow( <GifGrid category={ category } /> )
        expect( wrapper ).toMatchSnapshot()
    })
    
    test('should show items whe images are loading with useFetchGifs', () => {
        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cualquier cosa'
        },
        {
            id: 'ABCa',
            url: 'https://localhost/cualquier/cosa.jpg',
            title: 'Cualquier cosa'
        }]

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        })

        const wrapper = shallow( <GifGrid category={ category } /> )

        // expect( wrapper ).toMatchSnapshot()
        expect( wrapper.find('p').exists() ).toBe( false )
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length )
    })
})