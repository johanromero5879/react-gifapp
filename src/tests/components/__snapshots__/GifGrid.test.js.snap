// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing with <GifGrid /> should render correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    hug anime
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
